dbms/src/Storages/KVStore/MultiRaft/Disagg/FastAddPeer.cpp

```diff
@@ -185,26 +191,34 @@ std::variant<CheckpointRegionInfoAndData, FastAddPeerRes> FastAddPeerImplSelect(
     Stopwatch watch;
     std::unordered_map<StoreID, UInt64> checked_seq_map;
     auto fap_ctx = tmt.getContext().getSharedContextDisagg()->fap_context;
+    auto cancel_handle = fap_ctx->tasks_trace->getCancelHandleFromExecutor(region_id);
+
+    // Get candidate stores.
     const auto & settings = tmt.getContext().getSettingsRef();
-    auto current_store_id = tmt.getKVStore()->getStoreMeta().id();
-    auto candidate_store_ids = getCandidateStoreIDsForRegion(tmt, region_id, current_store_id);
+    auto current_store_id = tmt.getKVStore()->clonedStoreMeta().id();
+    std::vector<StoreID> candidate_store_ids;
+    fiu_do_on(FailPoints::force_set_fap_candidate_store_id, { candidate_store_ids = {1234}; });
+    if (candidate_store_ids.empty())
+        candidate_store_ids = getCandidateStoreIDsForRegion(tmt, region_id, current_store_id);
```

Comment:
maybe move the failpoint into `getCandidateStoreIDsForRegion`?

Reply:
fixed